messagebus/types/message.go

```diff
@@ -0,0 +1,18 @@
+package types
+
+type Message struct {
+	Topic   string
+	Key     string
+	ErrChan chan error
```

Comment:
I'm a little confused about the purpose of `ErrChan`, to make synchronized calls?

Reply:
Yes，if you want to wait for the message is sent to the broker, you should  set the ErrChan and wait for the response. And you can refer to messagebus/tests for the usage.

Reply:
I believe the most common use case would be asynchronous, say application layer code just fire a message and forget about it. So 1) we might need some kind of error handling, like to resend failed message inside `messagebus` lib, as you did in PR#103  2) the benchmark `BenchmarkSendOneMsg` tests about the synchronous case and could be considered as a lower-bound performance.

Reply:
For async usage, just leave the ErrChan to nil and don't wait for ErrChan to finish.
1) confluent-kafka-go already has retry mechanism for sending message. Currently, we set 5000ms timeout when sends a message, that means if the network has some problems, the message will be delivered failed after 5s.
2) Yes, for benchmark, we need to measure what time a single message needs to be delivered 'successfully'.

Reply:
Cool, thanks for the explanation!